### ISSUE TITLE
Regression: no completion for static fields and methods in some cases

### ISSUE THREAD

**Issue by 0dinD** [redhat-developer/vscode-java#2160]
In some cases (see example below), no completion is offered for static fields and methods, only for types. This appears to be a regression introduced between version 0.79.2 and 0.80.0 of the extension.

##### Environment
- Java extension version: >= 0.80.0

##### Steps To Reproduce
Have the following code:

```java
package com.example;

public class Foo {

	public Foo() {
		System. // Completion works fine here
		toString();

		new Runnable() {
			@Override
			public void run() {
				System. // Only suggests types
				toString();
				System. // Works fine
			}
		};

		new Runnable() {
			@Override
			public void run() {
				System. // Works fine
				this.toString();
				System. // Works fine
			}
		};
	}

}

```

##### Current Result
No completion is offered for static fields and methods in the case outlined above.

##### Expected Result
Completion should be offered there, as was the case before version 0.80.0 of the extension.

##### Additional Information
As mentioned, this seems to be a regression, because I can't reproduce this issue by installing version 0.79.2 or earlier of the extension. This in turn means that the regression lies between version 1.1.2 and 1.2.0 of eclipse.jdt.ls. I've also been able to reproduce this issue in the Eclipse IDE, version 2021-09. So it seems like this regression got introduced further upstream (see [the target platform changes between 1.1.2 and 1.2.0](https://github.com/eclipse/eclipse.jdt.ls/compare/v1.1.2...v1.2.0#diff-f15305031e24556c35cc42f0684bf5bbe9451e76b57735163e51652610367dfb)).

**Comment by snjeza** [comment]
>  can you take a look please?

@fbricon I can't reproduce the issue. I have created https://github.com/snjeza/vscode-test/raw/master/java-0.83.2.vsix

**Comment by 0dinD** [comment]
I just tried a fresh install of Eclipse 2021-12 M1, and still cannot reproduce the issue. Maybe it is a jdt.ls issue? But I was fairly certain that i reproduced it with my previous install of Eclipse 2021-09...

**Comment by snjeza** [comment]
@0dinD @fbricon  I have tried the following:

0.83.2
![assign](https://user-images.githubusercontent.com/388609/136866976-7401b441-e813-479d-94b7-9c58e80364dd.gif)

0.82.0
![assign2](https://user-images.githubusercontent.com/388609/136867207-426a0b8a-4a53-4854-a2a9-4d98d4f09b69.gif)

**Comment by snjeza** [comment]
The issue has been fixed in vscode-java >= 1.1.0
